datasource/ethereum/src/block_stream.rs

```diff
@@ -104,7 +82,7 @@ enum ReconciliationStep {
 /// The result of performing a single ReconciliationStep.
 enum ReconciliationStepOutcome {
     /// These blocks must be processed before reconciliation can continue.
-    YieldBlocks(Box<dyn Stream<Item = EthereumBlockWithTriggers, Error = Error> + Send>),
+    YieldBlocks(Vec<EthereumBlockWithTriggers>),
```

Comment:
Is there a good reason why `YieldBlocks` and `ProcessingDescendantBlocks` take a `Vec` but `YieldingBlocks` takes a `VecDeque`?

Reply:
I just avoided the conversion until the point where it's actually needed. We need a `VecDeque` at some point because we want to pop from the front, like a queue. Another possibility is to sort the blocks backwards when returning them from `blocks_with_triggers`.